torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -931,3 +931,156 @@ def call_triton_kernel(self, variable, args, kwargs, tx):
+        >>> @triton.jit
+        >>> def add_kernel(
+        >>>     in_ptr0,
+        >>>     in_ptr1,
+        >>>     out_ptr,
+        >>>     n_elements,
+        >>>     BLOCK_SIZE: "tl.constexpr",
+        >>> ):
+        >>>     pid = tl.program_id(axis=0)
+        >>>     block_start = pid * BLOCK_SIZE
+        >>>     offsets = block_start + tl.arange(0, BLOCK_SIZE)
+        >>>     mask = offsets < n_elements
+        >>>     x = tl.load(in_ptr0 + offsets, mask=mask)
+        >>>     y = tl.load(in_ptr1 + offsets, mask=mask)
+        >>>     output = x + y
+        >>>     tl.store(out_ptr + offsets, output, mask=mask)
+        >>>
+        >>> def add(x, y):
+        >>>     output = torch.empty_like(x)
+        >>>     n_elements = output.numel()
+        >>>
+        >>>     def grid_fn(meta):
+        >>>         return (triton.cdiv(n_elements, meta["BLOCK_SIZE"]),)
+        >>>
+        >>>     capture_triton(add_kernel)[grid_fn](x, y, output, n_elements, 16)
+        >>>     return output
+        >>>
+        >>> x = torch.randn(3, device="cuda")
+        >>> y = torch.randn(3, device="cuda")
+        >>> gm = make_fx(add)(x, y)
+        >>> print(gm.code)
+        >>> # def forward(self, x_1, y_1):
+        >>> #     empty_like = torch.ops.aten.empty_like.default(x_1, pin_memory = False)
+        >>> #     triton_kernel_wrapper_mutation_proxy = triton_kernel_wrapper_mutation(
+        >>> #         kernel_idx = 0, constant_args_idx = 0,
+        >>> #         grid = [(1, 1, 1)], kwargs = {
+        >>> #             'in_ptr0': x_1, 'in_ptr1': y_1, 'out_ptr': empty_like,
+        >>> #             'n_elements': 3, 'BLOCK_SIZE': 16
+        >>> #         })
+        >>> #     return empty_like
+
+    """
+    from triton.runtime.autotuner import Autotuner
+    from triton.runtime.jit import JITFunction
+
+    if not isinstance(triton_kernel, (JITFunction, Autotuner)):
+        raise RuntimeError(
+            "capture_triton only works on functions annotated with triton.jit or triton.autotune"
+        )
+    return TraceableTritonKernelWrapper(triton_kernel, None, None)
+
+
+from ..fx._symbolic_trace import is_fx_tracing
+
+
+class TracingTritonHOPifier(TritonHOPifier):
+    def raise_unsupported(self, msg):
+        raise RuntimeError(msg)
+
+    def is_callable(self, maybe_callable):
+        return callable(maybe_callable)
+
+    def call_grid(self, grid, meta, tx):
+        assert tx is None
+        return grid(meta)
+
+    def check_grid(self, grid):
+        if not isinstance(grid, tuple):
+            raise RuntimeError(
+                "capture_triton can only handle grids that resolve to Tuple[int, ...]."
+            )
+        return grid
+
+    def call_HOP(self, variable, grids, combined_args, tx):
+        assert tx is None
+        import torch
+
+        non_graphable_args = {
+            k: v
+            for k, v in combined_args.items()
+            if not isinstance(v, (torch.Tensor, int, float, bool))
```

Comment:
how did you come up with this list? pretty sure string also belongs in it

Reply:
From memory, I can add string to it. I'll go digging to see if there's a helper function for this in torch.fx already

Reply:
Side table in this file also contains a check for this, maybe use the same one as that one?

Reply:
Ah ok, I did that because triton.dtype was passed as an arg and we could not put that on the fx graph

Reply:
torch.fx has a way for checking this -- they encode the acceptable types in BaseArgumentType. I'll use that check